ros_ws/Dockerfile

```diff
@@ -0,0 +1,85 @@
+ARG ROS_DISTRO=noetic
+FROM ros:${ROS_DISTRO} as base-dependencies
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt update && apt install -y \
+  autoconf \
+  automake \
+  build-essential \
+  curl \
+  g++ \
+  gdb \
+  iputils-ping \
+  libboost-all-dev \
+  libeigen3-dev \
+  libtool \
+  libssl-dev \
+  make \
+  mesa-utils \
+  python3-setuptools \
+  python3-pip \
+  python3-rosdep \
+  python3-rosinstall \
+  python3-rosinstall-generator \
+  python3-wstool \
+  ros-${ROS_DISTRO}-rviz \
```

Comment:
What ros packages are preinstalled with this image? I'm thinking mainly about ros-control and ros-controllers...
Also, I would propose to add `ros-distro-joint-state-publisher-gui` here if it's not installed already because I find this is quite an important package and should be there by default.

Reply:
Ah nevermind, I just realized from your example that this is really just a base image and you will install whatever you want in the Dockerfile in your project...